src/Pass/ImgTagTransformPass.php

```diff
@@ -133,6 +144,19 @@ protected function isSvg(\DOMElement $el)
     }
 
     /**
+     * Detects if the img is a 1x1 pixel. In that case we convert to <amp-pixel> instead of <amp-img>
+     * @param DOMQuery $el
+     * @return bool
+     */
+    protected function isPixel(DOMQuery $el)
+    {
+        if (!$this->setResponsiveImgHeightAndWidth($el)) {
```

Comment:
My fault - should've re-read your code. The problem lies somewhere else: you duplicate the calculation of the image dimensions in convertAmpImg and convertAmpPixel. You could avoid this by moving this at the beginning instead:

```
$hasWidthAndHeight = $this->setResponsiveImgHeightAndWidth($new_el);
if (!$hasWidthAndHeight) {
  $this->addActionTaken(new ActionTakenLine('img', ActionTakenType::IMG_COULD_NOT_BE_CONVERTED, $lineno, $context_string));
  return false
}
if (isPixel(...)) {
  ...
}
```